src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -3196,10 +3196,12 @@ void MacroAssembler::compiler_fast_lock_object(Register oop, Register box, Regis
   Register temp = temp2;
   NearLabel done, object_has_monitor;
 
+  assert_different_registers(temp1, temp2, Z_R1_scratch);
+
   BLOCK_COMMENT("compiler_fast_lock_object {");
 
   // Load markWord from oop into mark.
-  z_lg(displacedHeader, 0, oop);
+  z_lg(displacedHeader, oopDesc::mark_offset_in_bytes(), oop);
 
   if (DiagnoseSyncOnValueBasedClasses != 0) {
     load_klass(Z_R1_scratch, oop);
```

Comment:
@RealLucy if we use `temp` here instead of Z_R1, do you think there will be issues ? It seems temp is free at this point.

Reply:
Using temp here actually is a good idea. As you know, using the scratch registers (Z_R0 and Z_R1) across calls is risky. You need to know exactly if they are used as scratch further down the call hierarchy.

Reply:
done.